### PR TITLE
WIP: Add basic run as overlay functionality

### PR DESCRIPTION
**Pull request by jzulauf-lunarg**
Add option to run as overlay, with no other changes.

Next steps, remove background when running as overlay.

When running with "overlay as service" https://github.com/LunarG/OpenXR-OverlayLayer/pull/25  note update artifacts.

### CODE REVIEW

File: src/tests/hello_xr/openxr_program.cpp [hunk 1]
```diff
@@ -581,6 +599,14 @@ struct OpenXrProgram : IOpenXrProgram {
             XrSessionCreateInfo createInfo{XR_TYPE_SESSION_CREATE_INFO};
             createInfo.next = m_graphicsPlugin->GetGraphicsBinding();
             createInfo.systemId = m_systemId;
+
+            XrSessionCreateInfoOverlayEXTX overlayCreateInfo = {XR_TYPE_SESSION_CREATE_INFO_OVERLAY_EXTX, nullptr, 0U, 0U};
```
Comment (utzcoz) [inline src/tests/hello_xr/openxr_program.cpp:603]: I remember OpenXR spec says the `next` of `XrSessionCreateInfoOverlayEXTX` can be `nullptr`, but the overlay will not show correct content on Android when `next` is `nullptr` with latest monado. In monado xrgears' new overlay PR, it uses graphic binding as `next` of overlayCreateInfo (https://gitlab.freedesktop.org/monado/demos/xrgears/-/merge_requests/10/diffs#575c3212e3fa074fbb40562ade4bdfad41bd74d7_593_598). Maybe hello_xr should use `m_graphicsPlugin->GetGraphicsBinding()` replace `nullptr` too.

Comment (ChristophHaag) [reply src/tests/hello_xr/openxr_program.cpp:603]: Yea both the graphics binding and XrSessionCreateInfoOverlayEXTX need to be in the next chain of XrSessionCreateInfo. Note that the validation layer has an issue there #264.

File: src/tests/hello_xr/openxr_program.cpp [hunk 2]
```diff
@@ -224,6 +233,10 @@ struct OpenXrProgram : IOpenXrProgram {
         std::transform(graphicsExtensions.begin(), graphicsExtensions.end(), std::back_inserter(extensions),
                        [](const std::string& ext) { return ext.c_str(); });
 
+        if (m_overlayApp) {
+            extensions.push_back(XR_EXTX_OVERLAY_EXTENSION_NAME);
```
Comment (utzcoz) [inline src/tests/hello_xr/openxr_program.cpp:237]: Before adding overlay extension as enabled extensions list, should it need to check whether runtime supports this extension likes xrgears(https://gitlab.freedesktop.org/monado/demos/xrgears/-/merge_requests/10/diffs#575c3212e3fa074fbb40562ade4bdfad41bd74d7_130_136)?

File: src/tests/hello_xr/main.cpp [hunk 3]
```diff
@@ -35,12 +35,13 @@ void ShowHelp() {
     // TODO: Improve/update when things are more settled.
     Log::Write(Log::Level::Info,
                "HelloXr --graphics|-g <Graphics API> [--formfactor|-ff <Form factor>] [--viewconfig|-vc <View config>] "
-               "[--blendmode|-bm <Blend mode>] [--space|-s <Space>] [--verbose|-v]");
+               "[--blendmode|-bm <Blend mode>] [--space|-s <Space>] [--overlay <Overlay placement>] [--verbose|-v]");
```
Comment (utzcoz) [inline src/tests/hello_xr/main.cpp:38]: Is there any chance to support those parameters for Android? Monado has two flavors, out of process and in process, and we can build two versions with different package name for different usage. Could hello_xr support similar flags to build two versions with different package name, one for normal usage, and one for overlay usage?

**Pull request closed without merging**